### PR TITLE
Increase Maximum Vblank Rate and Clocks Scale

### PR DESCRIPTION
**Pull request by elad335**
Allow x30 times the speed of vblank rate + clocks scale of original PS3.
In theory a 60 fps limit game which scales frame limit perfectly with vblank rate can be played at up to 1800 fps with this change.

And:
* Fixed lv2 sleep with Clocks Scaling.
* Make these settings dynamicaly adjustable.

### CODE REVIEW

**Comment by elad335** [comment]
Added a commit to make these setting dynamically adjustable as well.

**Comment by Megamouse** [comment]
Are you sure that every occurance of those settings is actually written with the dynamic property in mind?

**Comment by elad335** [comment]
Fixed a bug with clocks scaling 🤔.

**Comment by Megamouse** [comment]
maybe also check if the settings you made dynamic last time were also actually implemented dynamically 🤔 
I didn't really think about it back then.

File: rpcs3/Emu/RSX/RSXThread.cpp [hunk 1]
```diff
@@ -537,6 +537,8 @@ namespace rsx
 					while (get_system_time() - start_time >= period_time);
 
 					thread_ctrl::wait_for(wait_sleep);
+					period_time = 1000000 / g_cfg.video.vblank_rate; // update its' value
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/RSXThread.cpp]: 🤔 
``` 
// update its' value
```

**Comment by elad335** [comment]
Yea they were implemented correctly for dynamic updating.

File: rpcs3/Emu/RSX/RSXThread.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/Emu/RSX/RSXThread.cpp]: for dynamic setting changes.

**Comment by kd-11** [comment]
I struggle to understand why this PR exists. Outside of academic reasons, I don't see how someone was actually inconvenienced by not being able to run a game at a theoretical 1000+fps. I mean, anything over ~240 is already pointless so what's the motivation here?

**Comment by elad335** [comment]
Theoretical limits for benchmarking, in menues for example it actually makes sense to allow such frame rates.

**Comment by Asinin3** [comment]
Ahem @kd-11 
rpcs3spuelfvdank_PvaNHNQ2mT

**Comment by kd-11** [comment]
I still don't see it as being practical outside of looking at some numbers but hey, if there's real world demand for it its fine I guess.

**Comment by kd-11** [comment]
However, bear in mind these large values will interfere with the GUI sliders, unless the slider range is decoupled from the config range.

**Comment by elad335** [comment]
I'm aware, its nowhere near rsx wake-up delay limits so it should be fine.

File: rpcs3/Emu/RSX/RSXThread.cpp [hunk 2]
```diff
@@ -502,8 +502,8 @@ namespace rsx
 #endif
 			u64 start_time = get_system_time();
 
-			const u64 period_time = 1000000 / g_cfg.video.vblank_rate;
-			const u64 wait_sleep = period_time - u64{period_time >= host_min_quantum} * host_min_quantum;
+			u64 period_time = 1000000 / g_cfg.video.vblank_rate;
```
Comment (kd-11) [inline rpcs3/Emu/RSX/RSXThread.cpp]: If you need to update this every cycle then move the declaration inside the loop [here](https://github.com/RPCS3/rpcs3/pull/7331/files#diff-646d2875fee53ff3f6de5b4a68c7adc5R511) and avoid duplicate code.

**Comment by Asinin3** [comment]
Can a keybind be added to toggle vblank back to it's default state? Some games e.g have cutscenes that run too fast with it.

**Comment by elad335** [comment]
P.S. when squash merging do not forget to keep the commit description of the first commit!

**Pull request merged**